march_hardware/include/march_hardware/joint.h

```diff
@@ -64,16 +64,17 @@ class Joint
   double getVelocityIUAbsolute();
   double getVelocityIUIncremental();
   float getTemperature();
-  IMotionCubeState getIMotionCubeState();
+  MotorControllerStates getMotorControllerStates();
+  virtual bool checkMotorControllerState(std::ostringstream& error_stream);
```

Comment:
You could build this function into the `MotorControllerStates` object. This separates it from the `Joint` class, which clearly has nothing to do with any of this. Also I would separate the error stream getting from the check method in order to avoid returning variables through references.

Reply:
Split checkMotorControllerState to check and getErrorStatus, and moved both methods to abstract methods in MotorControllerStates. These are then implemented by a child struct IMotionCubeStates